src/WebJobs.Extensions.DurableTask/Listener/TaskOrchestrationShim.cs

```diff
@@ -56,6 +56,14 @@ public void SetFunctionInvocationCallback(Func<Task> callback)
                 FunctionType.Orchestrator,
                 this.context.IsReplaying);
 
+            await this.config.LifeCycleTraceHelper.OrchestratorStartingAsync(
```

Comment:
Done. However we also change the this part.  DurableOrchecturatorContext. We used to initialize the deferedFunctions Array in `AddDeferredTask`. We move it to the constructor. We want to know the lifecycle of the DurableOrchestratorContext. Is the instance created on each reply? 

```
        public override void AddDeferredTask(Func<Task> function)
        {
            this.deferedFunctions.Add(function);
        }

        public override async Task RunDeferredTasks()
        {
            if (this.deferedFunctions == null)
            {
                throw new InvalidOperationException(
                    "AddDeferredTask needs to be called first.");
            }
            await Task.WhenAll(this.deferedFunctions.Select(x => x()));
            this.deferedFunctions.Clear();
        }
```

Reply:
You should not assume that we create a new `DurableOrchestrationContext` for each reply. Yes, we create one today, but I have another optimization in another branch that causes it to not get recreated in some cases. I think your current implementation is good. However, you can get rid of the `null` check since `this.deferedFunctions` will never be null. :)